### PR TITLE
add uid/gid and target port

### PR DESCRIPTION
**Pull request by TaichiHo**
Add a couple annotations for an upcoming istio/cni change.
This change will enable us to control cni behavior over kubernetes annotation.
* targetPort
* proxyUID
* proxyGID

### CODE REVIEW

**Comment by howardjohn** [comment]
Can you give a little more info about why we would want this? I don't see why this needs to be customizable

**Comment by TaichiHo** [comment]
Thanks for the reply.

We are planning to use istio/cni without using istio. 

This change would allow us to redirect the traffic to the port our envoy is listening on. Same applies to uid and gid.

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -267,6 +267,26 @@ annotations:
     hidden: false
     resources:
       - Pod
+  - name: traffic.sidecar.istio.io/targetPort
+    description: The port where envoy listens for all TCP traffic
+    deprecated: false
+    hidden: false
+    resources:
+      - Pod
+  - name: traffic.sidecar.istio.io/noRedirectUID
```
Comment (costinm) [inline annotation/annotations.yaml:276]: How about 'sidecarGID'/'sidecarUID' ? Or 'noCaptureUID/noCaptureGID' ? 

Redirect happens to be what we use now - may change in future, and users may not understand it.

Also: do we still need both UID and GID ? I thought we found a way to use only GID ( which has the benefit of  allowing other UIDs to be excluded ).

File: annotation/annotations.yaml [hunk 2]
```diff
@@ -267,6 +267,26 @@ annotations:
     hidden: false
     resources:
       - Pod
+  - name: traffic.sidecar.istio.io/targetPort
```
Comment (costinm) [inline annotation/annotations.yaml:270]: Except we now have 2 ports ? Generic words like 'target' should be avoided - this is not the only target in the world or on 'traffic'. 

captureInPort/captureOutPort ( or ingress/egeress ) maybe ?

**Comment by TaichiHo** [comment]
Thanks @costinm 
Is there an existing RFC I can contribute to? I took a look at the team drive but didn't find one yet.

I might need to take some time to learn the history and understand the other use cases to come up with a proper RFC

**Pull request closed without merging**